### PR TITLE
Allow use of changelog feature without APACK

### PR DESCRIPTION
**Pull request by fabianlupa**
... by introducing a version constant setting in .abapgit.xml.

Draft. Is this something we want? I think the changelog feature might be something we want to offer without use of APACK and without being restricted to abapGit itself.

This does not solve the issue of the old version value being present after refresh as the load will not be invalidated / refreshed without a restart of abapGit (would need something like #5285 / issue #5284). One could of course directly specify the version in XML but then it's not accessible in ABAP for other uses the repo might itself have.

image

CC @sandraros @AndreaBorgia-Abo @bigld

### CODE REVIEW

File: src/zcl_abapgit_version.clas.abap [hunk 1]
```diff
@@ -242,4 +249,27 @@ CLASS zcl_abapgit_version IMPLEMENTATION.
...
 
   ENDMETHOD.
+
+  METHOD get_version_constant_value.
+    DATA: lv_version_class     TYPE string,
+          lv_version_component TYPE string.
+    FIELD-SYMBOLS: <lv_version> TYPE string.
+
+    IF iv_version_constant NP '*=>*'.
+      zcx_abapgit_exception=>raise( 'Version constant needs to use the format CLASS=>CONSTANT' ).
```
Comment (larshp) [inline src/zcl_abapgit_version.clas.abap:259]: guess it will also work with interfaces?

**Comment by larshp** [comment]
overall: nice, I like this solution

**Comment by larshp** [comment]
ref https://github.com/abapGit/abapGit/issues/3152

File: src/zcl_abapgit_version.clas.abap (hunk 1 above)
Comment (fabianlupa) [reply src/zcl_abapgit_version.clas.abap:259]: Yes it does (see change in abapGit's own .abapGit.xml . I couldn't think of a good way to account for this in the message and also the UI hint. `OBJECT` to me rather refers to instances and `{ CLASS / INTERFACE }=>CONSTANT` might just confuse. Maybe `CLASS=>CONSTANT / INTERFACE=>CONSTANT` would be better.

Comment (fabianlupa) [reply src/zcl_abapgit_version.clas.abap:259]: Well actually attributes will also work. So `COMPONENT` would be more correct than `CONSTANT`. Attributes will also call `class_constructor` though (do constants do that as well?) which means code execution of the (active) repo code :/.

**Comment by AndreaBorgia-Abo** [comment]
ping me when you need testing / review :)

File: src/zcl_abapgit_version.clas.abap (hunk 1 above)
Comment (mbtools) [reply src/zcl_abapgit_version.clas.abap:259]: class constants do not trigger the class constructor. reading class attributes does (on 754)

**Comment by mbtools** [comment]
Maybe add a read-only field below that shows the current value of the selected constant (assuming the code hasn't changed).

**Comment by mbtools** [comment]
- Remove unit test that doesn't work with transpiler
- Added read-only fields for version value:
 
image

Let's get this merged 👍

**Review by mbtools — APPROVED** [review]
LGTM 👍

**Review by larshp — APPROVED** [review]
lgtm

**Pull request merged**